services/gundeck/src/Gundeck/Push/Websocket.hs

```diff
@@ -130,10 +132,13 @@ bulkSend' uri (encode -> jsbody) = do
   where
     submit req = recovering (limitRetries 1) rpcHandlers $ const (rpc' "cannon" req id)
 
-    check req = req { Http.checkResponse = \rq rs ->
-        when (responseStatus rs /= status200) $
+    check loggr req = req { Http.checkResponse = \rq rs ->
```

Comment:
Let's remove these log lines and handle the `BulkPushResponse` more gracefully; i.e., when we call [bulkPush](https://github.com/wireapp/wire-server/blob/3d0645c8df417d9e739c7fc3455750dda205474a/services/gundeck/src/Gundeck/Push/Websocket.hs#L50-L65) let's ensure that the different response codes are handled independently

Reply:
I think this is not the case where some notifications in the `BulkPushRequest` have failed to be delivered, but the case where something went wrong with the entire request, e.g. "cannon down".

If we get a `BulkPushResponse`, it contains the status codes for the individual notifications, and those will be processed outside of `bulkSend'`.

If that's not the answer, please clarify!  :-)

Reply:
> If we get a BulkPushResponse, it contains the status codes for the individual notifications, and those will be processed outside of bulkSend'.

True. Still, I was thinking that we should probably handle a 400 (for instance) differently from other response codes?

Reply:
I'm not sure what the difference between 5xx, 400, or 339 should be.  We expect 200, and everything else should be handled gracefull, but also treated as a problem.  In fact, perhaps we should make the warn-level logging an err-level logging?

Otherwise I'm happy to distinguish the cases, but I'll need more requirements then.  Or a piece of code where we already get it right.

Reply:
after some off-line communication: 2d291b6